solid/src/main/java/com/inrupt/client/solid/SolidContainer.java

```diff
@@ -82,9 +82,9 @@ public SolidContainer(final URI identifier, final Dataset dataset, final Metadat
      * @return the contained resources
      */
     public Set<SolidResource> getResources() {
-        final String container = normalize(getIdentifier());
         // As defined by the Solid Protocol, containers always end with a slash.
-        if (container.endsWith("/")) {
+        if (getIdentifier().getPath().endsWith("/")) {
+            final String container = normalize(getIdentifier());
```

Comment:
Newly, the condition is not normalized. This seems intentional.

But what happens with URIs of the form `x/.` and `x/y/..`?
Maybe we should normalize before checking whether it ends in a slash.

[Example](https://www.mycompiler.io/view/2W9PVVdf5DM)

Reply:
`getIdentifier` from `RDFSource` via `SolidRDFSource` is a `URI` and `getPath` is not normalized.

Reply:
I've updated this to normalize the URI path